app/commands/mentor/student_relationship/toggle_favorited.rb

```diff
@@ -6,14 +6,18 @@ class ToggleFavorited
       initialize_with :mentor, :student, :favorited
 
       def call
-        return unless Mentor::Discussion.between(mentor: mentor, student: student).exists?
+        return unless allowed?
 
         relationship = Mentor::StudentRelationship.create_or_find_by!(
           mentor: mentor,
           student: student
         )
         relationship.update_column(:favorited, favorited)
       end
+
+      def allowed?
```

Comment:
@iHiD I think the code will be cleaner if this lives within the `Mentor::StudentRelationship` model. I didn't do that right now though and opted for this change as I'll leave that for you to decide.

Reply:
OK, thanks. I'm ok with it here for now, but I agree that if we use it in multiple places it would be worth adding elsewhere :)